sycl/test-e2e/Basic/accessor/accessor.cpp

```diff
@@ -1304,5 +1315,25 @@ int main() {
     });
   }
 
+  {
+    int data = 123;
+    int result = 0;
+    {
+      sycl::buffer<int, 1> data_buf(&data, 1);
+      sycl::buffer<int, 1> res_buf(&result, 1);
+      sycl::queue queue;
+      queue
+          .submit([&](sycl::handler &cgh) {
+            implicit_conversion::ResAccT res_acc = res_buf.get_access(cgh);
+            implicit_conversion::AccT acc(data_buf, cgh);
+            cgh.single_task([=]() {
+              implicit_conversion::implicit_conversion(acc, res_acc);
+            });
+          })
+          .wait_and_throw();
+    }
+    assert(result == 123 && "Expected value not seen.");
+  }
```

Comment:
Could we maybe also have a test-case for `swap`?

Reply:
@steffenlarsen , done